### PR TITLE
 Migrate `projectile-test-ignored-directory-p' to buttercup 

### PR DESCRIPTION
**Pull request by kevinjfoley**
-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../CONTRIBUTING.md)
- [x] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [x] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

**Comment by kevinjfoley** [comment]
Only have one test in this PR to make sure I have the hang of things.

Also I think the test should be modified slightly because as is the regular expression covers both truthy test cases so we should change `/path/to/project/tmp` to be something that doesn't match the regex and test that.  I can add that to this PR if you agree or open a separate one if that's more appropriate.

**Comment by bbatsov** [comment]
Feel free to add this to the current PR. The old tests are pretty bad and should not be copied one to one.

File: test/projectile-test.el [hunk 1]
```diff
@@ -98,3 +98,10 @@ test temp directory"
           (projectile-known-projects-file (projectile-test-tmp-file-path)))
       (projectile-clear-known-projects)
       (expect projectile-known-projects :to-equal nil))))
+
+(describe "projectile-test-ignored-directory-p"
+  (it "ignores specified directories"
```
Comment (bbatsov) [inline test/projectile-test.el]: I think you can have a separate `if` for the regexp check to highlight it.

Comment (kevinjfoley) [reply test/projectile-test.el]: Good call, I added a separate spec for it

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**